### ISSUE TITLE
Infix extension function for doReturn stub construction 

### ISSUE THREAD

**Issue by RobertZagorski** [mockito/mockito-kotlin#258]
Mockito provides 2 constructions for stubbing:
`doReturn(...).when(...)`
`when(...).thenReturn(...)`
The first one is known not to call real method, which is sometimes handy ([source](https://stackoverflow.com/a/29394497/6507689))
There are nice extensions for the second use case, which allow for more concise writing of stubbing:
```
search.stub {
     on { what } doReturn WHAT
}
```

It would be nice to provide consise construction for the first use case:

```
search.stub {
     doReturn { WHAT } on { what}
}
```
Those examples are of course very basic, but it would become more helpful in more advanced use cases.
Basically what is needed is providing method `doReturn` function in `KStubbing` interface and infix extension function called `on` to `OngoingStubbing` interface, I guess. 

I hope you see my point

**Comment by glureau-betclic** [comment]
You can use lambda with doAnswer. Is it not enough?

```
search.stub {
    on { what } doAnswer { WHAT }
}
```
